### DISCUSSION TITLE
What is equivalent to `env FOO=value1 BAR=value2 command arg1 arg2` in Bash?

### DISCUSSION THREAD

**Discussion by tats-u** [PowerShell/PowerShell#19925] [Q&A] [answered]
Does PowerShell still have to split this task into the following 3 commands?

```pwsh
$env:FOO = "value1"
$env:BAR = "value2"
command arg1 arg2
```

**Comment by kasini3000** [comment, marked as answer]
$env:FOO = "value1" ; $env:BAR = "value2" ; command arg1 arg2

----

$foo = 1 ;$bar=2; command arg1 arg2

command:
```
if ($foo) {} else {$foo = 'default'}
...
```

**Reply by tats-u** [reply]
I couldn't understand the below of the horizontal line, but `env:` in `$env:FOO` is required, right?

**Reply by kasini3000** [reply]
```
./etcdctl.ps1 -ETCDCTL_API  3  -endpoints 'http://1.2.3.4:2379' 
```
Usually written as:
```
$h=@{ ETCDCTL_API=3 ;endpoints = 'http://1.2.3.4:2379' } ; ./etcdctl.ps1 @h
```

```
$foo = 1 ;$bar=2; command $foo $bar #powershell
foo=1 ; bar=2; command $foo $bar #shell
```

**Reply by tats-u** [reply]
You mean local variables, right?
I'm asking only environment variables.
It seems impossible unless split into multiple commands by `;`. Thanks.

**Reply by kasini3000** [reply]
> I'm asking only environment variables.

Global foo supports storing objects instead of storing string numbers like environment variables.
It is recommended to use [$global:foo=1] and multithreaded concurrent scripts for powershell scripts. 
Global foo can also be used in multiple sequential scripts.

Unlike shells, I do not recommend using environment variables unless they are shared by multiple processes.

**Reply by tats-u** [reply]
The target isn't made by PowerShell. We can't help it.

**Comment by stonehellway** [comment]
Hi,

Don't forget you can do that if the command accepts pipeline input:

($a, $b = 10, 5) | Sort-Object

**Reply by tats-u** [reply]
I mean how to _run a command_ with some _environment_ variable set in a _single_ command.
The command I want to run doesn't support that way.
